week-4/E-arrays-of-objects/exercise-3.js

```diff
@@ -59,14 +59,32 @@ var restaurantFinderApplication = {
   applicationVersion: "1.0",
   restaurants: restaurants,
   findAvailableRestaurants: function(numberOfPeople) {
+    function findRestaurant(restaurant) {
+      return (
+        restaurant.totalSeats - restaurant.numberOfCustomers > numberOfPeople
+      );
+    }
+    return restaurants.filter(findRestaurant).map(R => R.name);
+
     // Complete here
   },
   findRestaurantServingDish: function(dishName) {
+    function checkDish(dish) {
+      return restaurants.includes(dish);
```

Comment:
Nearly - you need to check the menu though. And passing a function to restaurants.filter will make it act on each restaurant, so what you really want is more like:
`function checkRestaurant(restaurant) { return restaurant.menu.includes(dishName); }`.